File: py/IEC61968/AssetInfo/ConcentricNeutralCableInfo.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.Length import Length
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.ResistancePerLength import ResistancePerLength

class ConcentricNeutralCableInfo:
    
    def __init__(self):
        self.diameter_over_neutral = Length()
        self.neutral_strand_count = 0
        self.neutral_strand_gmr = Length()
        self.neutral_strand_radius = Length()
        self.neutral_strand_rdc20 = ResistancePerLength()

File: java/IEC61968/AssetInfo/ConcentricNeutralCableInfo.java
package IEC61968.AssetInfo;

import IEC61970.Base.Domain.Length;
import IEC61970.Base.Domain.Integer;
import IEC61970.Base.Domain.ResistancePerLength;

/**
 * Concentric neutral cable data.
 * @author Tom
 * @version 1.0
 * @created 25-Dec-2023 8:45:20 PM
 */
public class ConcentricNeutralCableInfo extends CableInfo {

	/**
	 * Diameter over the concentric neutral strands.
	 */
	public Length diameterOverNeutral;
	/**
	 * Number of concentric neutral strands.
	 */
	public Integer neutralStrandCount;
	/**
	 * Geometric mean radius of the neutral strand.
	 */
	public Length neutralStrandGmr;
	/**
	 * Outside radius of the neutral strand.
	 */
	public Length neutralStrandRadius;
	/**
	 * DC resistance per unit length of the neutral strand at 20 øC.
	 */
	public ResistancePerLength neutralStrandRDC20;

	public ConcentricNeutralCableInfo(){

	}
}//end ConcentricNeutralCableInfo